### ISSUE TITLE
Deep Learning for Sequential Recommendation: Algorithms, Influential Factors, and Evaluations

### ISSUE THREAD

**Issue by zerebom** [zerebom/paper-books#55]
https://dl.acm.org/doi/fullHtml/10.1145/3426723

注意☠️ : まだかなり斜め読みなので、解釈が誤っているかもです。後半ほどまとめが雑です 🙏 

## 概要
Sequential Recommendationの包括的なサーベイ論文。50ページ近くある。以下のようなことがまとめてある。

- Sequential Recommendationとは何か
- Sequential Recommendationにはどんな種類があるか
- Sequential Recommendationのモデルの性能に影響を与える主要な要因は何か
- Sequential Recommendationの未解決課題は何か

## Introduction
- Recsysのおかげで情報過多の問題を解決できる
- 従来の推薦システムの多くは、コンテンツベースの協調フィルタリングにもとづく
   - ItemとUserの暗黙的な相互作用に基づいてユーザのアイテムに対する嗜好をモデル化する
- ただし、これらの手法は過去の履歴における全ての相互作用を等しく重要であると仮定し、静的な嗜好を学習する  
    - 実際には、最新のインタラクションほどユーザの嗜好には影響を与えるはず
    - また、逐次的な依存関係を無視されてしまう
- DLベースのSequential Recommendationなら、異なるセッションにまたがる長期的な嗜好も、あるセッションにおける興味も同時にモデル化できるよ

最近は研究が盛んになってるよ。
> ![image](https://user-images.githubusercontent.com/38466611/160859187-be0496da-da5f-49c2-a374-8a0630ebbb8e.png)

### Q. そもそもSequential Recommendationとは何か?
> ![image](https://user-images.githubusercontent.com/38466611/160857152-4d9fc46d-c073-4e58-ac1a-6b8dac227f88.png)

A. Behavior sequence(= trajectory)を入力とし、次のBehaviorを当てるタスクのこと。

#### 言葉の定義
- Behavior
  - あるユーザのあるアイテムに対するあるアクションのこと(ai)
  - それぞれ(ci, oi)とタプルで表せる
-  Behavior type(ci)
   - アイテムに対するアクションの種類
   - ex) search, click, add-to-cart, buy, share 
- Behavior object(oi)
   - ユーザが選んでインタラクトするもの。アイテムのこと 

### Q. Behavior sequenceにはどんな種類があるの?
A. 以下の3つに分けられるよ

#### Experience-based behavior sequence
- 設定: ユーザが単一のアイテムに対して多様なアクションをとる
- ゴール: あるユーザはこのアイテムに対してどんなアクションをとるのか?

> ![image](https://user-images.githubusercontent.com/38466611/160859380-d42398a7-deb4-4b74-b3c4-b91ea300461a.png)

#### Transaction-based behavior sequence
- 設定: ユーザが複数アイテムに対して単一なアクションをとる
- ゴール: あるユーザはあるアイテムに対してアクションXX(eg. buy) をとるのか?
    - (余談)→ H&Mコンペはこれ

> ![image](https://user-images.githubusercontent.com/38466611/160859414-d9640701-eab3-4c97-8a85-db88b0196c74.png)

#### Interaction-based behavior sequence
- 設定: ユーザが複数アイテムに対して多様なアクションをとる
- ゴール: あるユーザはあるアイテムに対してどんなアクションをとるのか?
> ![image](https://user-images.githubusercontent.com/38466611/160859457-52456e6d-2bce-4ea0-bb84-f3bcf549b2c7.png)

### Q. Sequential Recommendation Tasksにはどんな種類があるの?
A. 以下の2つがある
- next-item recommendation
   - 単一のbehaviorに対して単一のobjectが含まれる
- next-basket recommendation  
   - 単一のbehaviorに対して複数のobjectが含まれる

> ![image](https://user-images.githubusercontent.com/38466611/160860104-3dcae772-b044-4dd0-9706-c70e73353d8d.png)

Sequential Recommendation Tasksは、actionとuserを引数に次のアイテムの選択確率を算出する関数の学習と言い換えられる。
> ![image](https://user-images.githubusercontent.com/38466611/160860049-74a6a8de-1952-43ba-9d0e-33a9eaab81f6.png)

### 最近どんな研究が盛んなの?
A. 最近はAttentionベース * next-item recommendationが盛んみたい

- オレンジ: RNN
- 黄色: CNN
- 青: Attention
- 黄色: GNN

> ![image](https://user-images.githubusercontent.com/38466611/160860282-4196cd51-b655-45f7-96e3-47c72730a5fb.png)

### 3.2 Transaction-based Sequential Recommendation
(TBD) 既存のTransaction-based Sequential Recommendationの学習に使われているメソッドの紹介

GRU4Recが有名だよ。

### Q. どんなアーキテクチャがいいの?
A. いろんなGRU4Recを用意して実験した。

> ![image](https://user-images.githubusercontent.com/38466611/160862276-59bc516b-5efa-4a9d-bf3c-5efbf7195a19.png)

- u_id, c_idをそれぞれemb → GRUに入れてからconcat→FF→ Softmaxとするアーキテクチャが強そう

> ![image](https://user-images.githubusercontent.com/38466611/160862847-a1742492-0ae1-41f4-b84e-2b405ae97116.png)

### Q. Sequential Recommendationする上で何が学習に効くの?

前提: DL-based modelの学習パイプラインには以下の要素があるので、それぞれ見ていく。
![image](https://user-images.githubusercontent.com/38466611/160862994-381c6b3a-1ce4-4b36-a03f-e2c2ff139152.png)

A. 図の通り
> ![image](https://user-images.githubusercontent.com/38466611/160861047-3a6a3423-95cb-4a5a-bf44-b63c54ea42a9.png)

(かなり理解が怪しいが) 以下のことがわかる。
- Behavior間の経過時間(Dwell time)を入れると精度が上がる
- Attention機構を使うと少し精度が上がっている
- Userの特徴を入れる(category idとかで)実験はうまくいってなさそう

### Q. どんなLoss funcが良いの?
A. タスク・データセットによるが、 BPR-max, Cross Entropyあたりを使うのが良さそう。

> ![image](https://user-images.githubusercontent.com/38466611/160861953-906f7873-f0f8-45da-a6a6-bc837413b7a8.png)

**Comment by zerebom** [comment]
### Loss func の紹介
- L top1-max:
- L BPR-max
- CCE: log softmax
- hinge: 二値分類で

<img width="1463" alt="image" src="https://user-images.githubusercontent.com/38466611/163356632-9918d8af-d58f-44ff-878d-6449bfa20188.png">
